src/jit/flowgraph.cpp

```diff
@@ -22445,6 +22445,28 @@ GenTreePtr Compiler::fgInlinePrependStatements(InlineInfo* inlineInfo)
 
 void Compiler::fgInlineAppendStatements(InlineInfo* inlineInfo, BasicBlock* block, GenTreePtr stmtAfter)
 {
+    // If this inlinee was passed a generic context but didn't look at
+    // it, we can decrement the "generic context was used" ref count.
+    if ((inlineInfo->inlineCandidateInfo->methInfo.args.callConv & CORINFO_CALLCONV_PARAMTYPE) != 0)
+    {
+        // Did the context require the caller to perform a runtime lookup?
+        if (inlineInfo->inlineCandidateInfo->exactContextNeedsRuntimeLookup)
+        {
+            // Fetch the temp for the generic context
+            const unsigned typeCtxtArg = inlineInfo->typeContextArg;
+            const unsigned tmpNum      = inlineInfo->lclTmpNum[typeCtxtArg];
+
+            // Was it used in the inline body?
+            if (tmpNum == BAD_VAR_NUM)
+            {
+                // No, so the runtime lookup is not needed.
+                JITDUMP("Inlinee ignores runtime lookup generics context\n");
+                assert(lvaGenericsContextUseCount > 0);
+                lvaGenericsContextUseCount--;
```

Comment:
"collectible classes" are also known as dynamic methods and LCG (which in turn mean lightweight codegen)

It happens when a application creates a method dynamically at runtiume by specify it IL and then executing it.

they are "collectable" because when a full GC occurs it also marks each of these dynamic methods as either reachable (and thus alive) or unreachable and thus collected and the JIT code buffer and associated VM datastructures are put onto a free list to be reused (by a future dynamic method)

Reporting the 'this' pointer for such a method insures that the class and any methods that it contains are not collected by the GC.